src/net.h

```diff
@@ -725,6 +725,7 @@ class CNode
         int64_t nextSendTimeFeeFilter{0};
     };
 
+    // m_tx_relay == nullptr if we're not relaying transactions with this peer
```

Comment:
Isn't a boolean a better practice than a nullptr ? Do we prefer to have a node crashing because of non-initialized acces or a topology leak because of a boolean in wrong state ?

Reply:
Sorry for the noise on this one, I thought about it after the fact, maybe the ideal would be to have special classes CBlockNode/CTxNode but would need different code paths so that's for future works.